From b654484de707ab1c796a1a6ee3ce87d91a443da7 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Fri, 20 Apr 2018 14:44:18 +0800
Subject: [PATCH 5/5] Revert "video: fbdev: fsl: fix a deadlock from
 console_lock and blank.rwsem"

This reverts commit 6125aee3429866ff0bc9e152a8bbe8bac3e03f93.
This patch would cause the HDMI not working, so revert it and no deadlock found
yet on latest kernel version.

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/video/fbdev/fsl-sii902x.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/video/fbdev/fsl-sii902x.c b/drivers/video/fbdev/fsl-sii902x.c
index 3e08923..969f8ba 100644
--- a/drivers/video/fbdev/fsl-sii902x.c
+++ b/drivers/video/fbdev/fsl-sii902x.c
@@ -294,8 +294,9 @@ static void det_worker(struct work_struct *work)
 		}
 
 		/* make sure fb is powerdown */
-		if (fbi->fbops->fb_blank)
-			fbi->fbops->fb_blank(FB_BLANK_POWERDOWN, fbi);
+		console_lock();
+		fb_blank(fbi, FB_BLANK_POWERDOWN);
+		console_unlock();
 
 		if (monspecs->modedb_len > 0) {
 			int i;
@@ -326,13 +327,15 @@ static void det_worker(struct work_struct *work)
 			console_unlock();
 		}
 
-		if (fbi->fbops->fb_blank)
-			fbi->fbops->fb_blank(FB_BLANK_UNBLANK, fbi);
+		console_lock();
+		fb_blank(fbi, FB_BLANK_UNBLANK);
+		console_unlock();
 	} else {
 		sii902x->cable_plugin = 0;
 		sprintf(event_string, "EVENT=plugout");
-		if (fbi->fbops->fb_blank)
-			fbi->fbops->fb_blank(FB_BLANK_POWERDOWN, fbi);
+		console_lock();
+		fb_blank(fbi, FB_BLANK_POWERDOWN);
+		console_unlock();
 	}
 	kobject_uevent_env(&sii902x->client->dev.kobj,
 			KOBJ_CHANGE, envp);
-- 
1.8.5.2.233.g932f7e4

